test/smoke/go/kube.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	
+	clientcmd "k8s.io/client-go/tools/clientcmd"
+	restclient "k8s.io/client-go/rest"
+	
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/meta"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	
+	clientbuildv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+)
+
+const (
+	ENV_VAR_KUBECONFIG = "KUBECONFIG"
+)
+
+var (
+	defaultKubeconfigPath = filepath.Join(".kube", "config")
+	
+	kubeClient *client.Client
+)	
+
+func initKubeClient(){
+	if kubeClient == nil {
+		kubeClient = client.NewForConsole()
+		config, err := buildKubeConnectionConfig()
+		if err != nil {
+			panic(fmt.Sprintf("Error building Kube connection config: %v", err))
+		}
+		kubeClient.BuildCli, err = clientbuildv1.NewForConfig(config)
```

Comment:
I guess the `BuildCli` could be build in the `NewForConsole` function or a new function in that package, like `client.WithBuildConfig().NewForConsole()`

Reply:
Made some modifications on that part, I let you review it.

we can now do `client.newBuilder().With*().Build()`
I updated also `NewForConsole` and `NewForController` with this form.